src/api/app/views/webui2/shared/_sign_up.html.haml

```diff
@@ -6,14 +6,21 @@
 - else
   = form_tag({ controller: 'user', action: 'register', method: :post }, class: 'sign-up', autocomplete: 'off') do
     .form-group
-      = text_field_tag 'login', nil, placeholder: 'Username', autocomplete: 'off', class: 'form-control'
+      = label_tag 'login', 'Username:'
+      %abbr.text-danger{ title: 'required' } *
+      = text_field_tag 'login', nil, placeholder: 'Username', autocomplete: 'off', class: 'form-control', required: true
     .form-group
-      = text_field_tag 'email', nil, placeholder: 'Email address', autocomplete: 'off', class: 'form-control'
+      = label_tag 'email', 'Email:'
+      %abbr.text-danger{ title: 'required' } *
+      = text_field_tag 'email', nil, placeholder: 'Email address', autocomplete: 'off', class: 'form-control', required: true
```

Comment:
Let's set the type here to 'email'. That way we have a more specific form validation. Checkout https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation and https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation#Validation-related_attributes for more details:-)

https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation#Validation-related_attributes